### PR TITLE
[py-lvis] New package

### PR DESCRIPTION
**Pull request by qwertos**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-lvis/package.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    depends_on("py-cycler@0.10:", type=("build", "run"))
+    depends_on("py-cython@0.29.12:", type=("build", "run"))
+    depends_on("py-kiwisolver@1.1:", type=("build", "run"))
+    depends_on("py-matplotlib@3.1.1:", type=("build", "run"))
+    depends_on("py-numpy@1.18.2:", type=("build", "run"))
+    depends_on("opencv@4.1.0.25:+python3", type=("build", "run"))
+    depends_on("py-pyparsing@2.4.0:", type=("build", "run"))
+    depends_on("py-python-dateutil@2.8:", type=("build", "run"))
+    depends_on("py-six@1.12:", type=("build", "run"))
+    depends_on("py-pycocotools", type=("build", "run"))
```
Comment (manuelakuhn) [inline var/spack/repos/builtin/packages/py-lvis/package.py:30]: Where is this requirement coming from?

Comment (qwertos) [reply var/spack/repos/builtin/packages/py-lvis/package.py:30]:
```python
>>> import lvis
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/jchits/workspace/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-11.2.0/py-lvis-0.5.3-6tzocas6smv6xa6kne2bejwsluie3rgw/lib/python3.10/site-packages/lvis/__init__.py", line 2, in <module>
    from lvis.lvis import LVIS
  File "/home/jchits/workspace/spack/opt/spack/linux-rhel7-skylake_avx512/gcc-11.2.0/py-lvis-0.5.3-6tzocas6smv6xa6kne2bejwsluie3rgw/lib/python3.10/site-packages/lvis/lvis.py", line 15, in <module>
    import pycocotools.mask as mask_utils
ModuleNotFoundError: No module named 'pycocotools'
```

Comment (manuelakuhn) [reply var/spack/repos/builtin/packages/py-lvis/package.py:30]: Thanks. Can you add a comment  about that?

**Comment by qwertos** [comment]
@spackbot fix style

File: var/spack/repos/builtin/packages/py-lvis/package.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    depends_on("opencv@4.1.0.25:+python3", type=("build", "run"))
+    depends_on("py-pyparsing@2.4.0:", type=("build", "run"))
+    depends_on("py-python-dateutil@2.8:", type=("build", "run"))
+    depends_on("py-six@1.12:", type=("build", "run"))
+
+    # imported at lvis/lvis.py:15
+    depends_on("py-pycocotools", type=("build", "run"))
+
+    def patch(self):
+        ln = which("ln")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-lvis/package.py]: This won't work on Windows. Maybe better just to use something like `os.rename`

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Comment by qwertos** [comment]
@spackbot fix style

**Review by adamjstewart — DISMISSED** [review]
Could still use a comment like:
```python
# pycocotools is not listed as a dependency but is required to import
```

**Comment by qwertos** [comment]
@spackbot fix style

**Comment by qwertos** [comment]
@spackbot fix style

**Review by adamjstewart — DISMISSED** [review]
You can run `spack style --fix` from the command line to fix the style issues

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**